DRAFT_homogeneous_impact_areas.Rmd

```diff
@@ -0,0 +1,426 @@
+infra_agg_terrestrial <- st_crop(infra_agg, outline)
+saveRDS(infra_agg_terrestrial, "P:/41201785_okologisk_tilstand_2022_2023/data/cache/infra_agg_terrestrial.rds")
+```
+
+```{r, include=F}
+# silently load cached file
+infra_agg_terrestrial <- readRDS("P:/41201785_okologisk_tilstand_2022_2023/data/cache/infra_agg.rds")
+```
+
+### Discretize the entire map
+
+```{r}
+names(infra_agg_terrestrial) <- "infrastructureIndex"
+
+infra_agg_discrete <- infra_agg_terrestrial %>%
+  mutate(infrastructureIndex = case_when(
+    infrastructureIndex < 1 ~ 0,
+    infrastructureIndex < 6 ~ 1,
+    infrastructureIndex < 12 ~ 2,
+    infrastructureIndex >= 12 ~ 3
+  ))
+```
+
+### Vectorize
+
+This step might seem rather stupid.
+We want to vectorize a rather large raster.
+This makes it a quite big data object.
+The reason is that there is no really good way to burn polygon data on to raster grid cells after the disuse of the `{raster}` package.
+It was not straight forward then either.
+But calculating intersections between polygons is very fast and easy.
+
+```{r, warning=F}
+infra_agg_discrete_vect <- eaTools::ea_homogeneous_area(infra_agg_discrete, groups = infrastructureIndex)
+```
+
+Let's plot this now.
+First, lets crop it to reduce computation time.
+
+```{r}
+myBB <- st_bbox(c(xmin=260520.12, xmax = 278587.56,
+                ymin = 7032142.5, ymax = 7045245.27),
+                crs = st_crs(infra_agg_discrete_vect))
+
+infra_agg_discrete_vect_trd <- st_crop(infra_agg_discrete_vect, myBB)
+```
+
+```{r test-2, fig.cap="A vectorized version of the Infrastructure indeks over Trondheim on a four-step discrete scale."}
+tm_shape(infra_agg_discrete_vect_trd)+
+  tm_polygons(col="infrastructureIndex",
+              style = "cat")+
+  tm_layout(legend.outside = T)+
+  tm_shape(hw_utm)+
+  tm_lines(col="white")
+```
+
+As can be seen in the figure above, `st_warp` merges grid cells that have the same value, and the vectorized raster doesn't not end up being that big in the end.
+
+## Chech national distribution
+
+```{r}
+regions <- sf::read_sf("data/regions.shp", options = "ENCODING=UTF8")
+unique(regions$region)
+```
+
+```{r}
+st_crs(infra_agg_discrete_vect) == st_crs(regions)
+```
+
+
+Since the two layers are completely overlapping, we can get the intersections
+```{r, eval=F}
+infra_stats <- eaTools::ea_homogeneous_area(infra_agg_discrete_vect,
+                             regions,
+                             keep1 = "infrastructureIndex",
+                             keep2 = "region")
+saveRDS(infra_stats, "P:/41201785_okologisk_tilstand_2022_2023/data/cache/infra_stats.rds")
+```
+
+```{r, echo=F}
+# silently read cached file
+infra_stats <- readRDS("P:/41201785_okologisk_tilstand_2022_2023/data/cache/infra_stats.rds")
+```
+
+Let's calculate the areas of these polygons and compare the HIF in the five regions. 
+```{r HIF-region, fig.cap="Stacked barplot showing the distribution of human impact factor across five regions in Norway."}
```

Comment:
For the first graph, you may consider using km^2 as a unit, instead of 1e6 m^2, but that's merely cosmetic.
However, the second figure is not the area (m^2) but the fraction, right? The x-axis legend needs to show that.

Reply:
Very good. Fixed.